climakitae/data_export.py

```diff
@@ -0,0 +1,225 @@
+import xarray as xr
+import pandas as pd
+import shutil
+import dask
+import warnings
+import datetime
+import numpy as np
+xr.set_options(keep_attrs=True)
+import os
+
+def _export_to_user(user_export_format,data_to_export,
+                    file_name,**kwargs):
+    """
+    The data export method, called by core.Application.export_dataset. Saves
+    a dataset to the current working directory in the output format requested by the user 
+    (which is stored in 'user_export_format').
+    
+    user_export_format: pulled from dropdown called by app.export_as()
+    data_to_export: xarray ds or da to export
+    variable_name: string corresponding to variable to export
+    file_name: string corresponding to desired output file name
+    kwargs: variable, scenario, and simulation (as needed)
+    
+    NOTE: requires the following at the top of the notebook:
+    !pip install pytest-shutil
+    """
+    
+    excel_row_limit = 1048576
+    path_and_name = './'+file_name
+    req_format = user_export_format.output_file_format
+    ds_attrs = data_to_export.attrs
+    ct = datetime.datetime.now()
+    ct_str = ct.strftime("%d-%b-%Y (%H:%M:%S)")    
+    ck_attrs = {'data_retrieved_by' : 'Cal-Adapt Analytics Engine v 0.0.1',
```

Comment:
add note to backlog to update this line either automatically, or every time we push across a new version manually....